sdk/storage/Azure.Storage.Blobs/src/BlobServiceClient.cs

```diff
@@ -41,6 +41,19 @@ public class BlobServiceClient
         /// </summary>
         internal virtual HttpPipeline Pipeline => _pipeline;
 
+        /// <summary>
+        /// The <see cref="BlobClientOptions"/> used to make this client's <see cref="Pipeline"/>.
+        /// </summary>
+        private readonly BlobClientOptions _sourceOptions;
+
+        /// <summary>
+        /// A deep copy of the <see cref="BlobClientOptions"/> used to make this client. Every call to this property
+        /// will return a new deep copy of the original, free to safely mutate.
+        /// </summary>
+        internal virtual BlobClientOptions SourceOptions => _sourceOptions == default
+            ? default
+            : new BlobClientOptions(_sourceOptions);
```

Comment:
We shouldn't create new instances every time the property is accessed.

Reply:
We should. This is a mutable class. Allowing ourselves to edit this breaks assumptions about what the field represents.

However, without the Azure.Core changes, we can't create a new instance anymore, so we're going to have to live with mutating these values for now.